src/projectscene/view/clipsview/clipslistmodel.cpp

```diff
@@ -476,43 +496,86 @@ bool ClipsListModel::trimRightClip(const ClipKey& key, bool completed)
 
 void ClipsListModel::selectClip(const ClipKey& key)
 {
-    selectionController()->setSelectedClip(key.key);
-    selectionController()->setSelectedTracks(TrackIdList({ key.key.trackId }));
+    Qt::KeyboardModifiers modifiers = keyboardModifiers();
+
+    if (modifiers.testFlag(Qt::ControlModifier)) {
+        appendToSelectedClips(key);
+    } else {
+        if (muse::contains(selectionController()->selectedClips(), key.key)) {
+            return;
+        }
+        selectionController()->setSelectedClips(ClipKeyList({ key.key }));
+        selectionController()->setSelectedTracks(TrackIdList({ key.key.trackId }));
+    }
+}
+
+void ClipsListModel::appendToSelectedClips(const ClipKey &key)
+{
+    auto selectedClips = selectionController()->selectedClips();
+    auto selectedTracks = selectionController()->selectedTracks();
+
+    if (!muse::contains(selectedClips, key.key)) {
+        selectedClips.push_back(key.key);
+    }
+    if (!muse::contains(selectedTracks, key.key.trackId)) {
+        selectedTracks.push_back(key.key.trackId);
+    }
+
+    selectionController()->setSelectedClips(selectedClips);
+    selectionController()->setSelectedTracks(selectedTracks);
```

Comment:
better move method `appendToSelectedClips` to `selectionController` (add new method and, I think, rename to `addSelectedClips`)